lib/openhab/dsl/items/image_item.rb

```diff
@@ -0,0 +1,164 @@
+        #
+        # @param [String] file location
+        # @param [String] mime_type of image
+        #
+        #
+        def update_from_file(file, mime_type: nil)
+          file_data = IO.binread(file)
+          mime_type ||= (MimeMagic.by_path(file) || MimeMagic.by_magic(file_data))&.type
+          update_from_bytes(file_data, mime_type: mime_type)
+        end
+
+        #
+        # Update image from image at URL
+        #
+        # @param [String] uri location of image
+        #
+        #
+        def update_from_url(uri)
+          logger.debug("Downloading image from #{uri}")
+          response = Net::HTTP.get_response(URI(uri))
+          mime_type = response['content-type']
+          bytes = response.body
+          mime_type ||= detect_mime_from_bytes(bytes: bytes)
+          update_from_bytes(bytes, mime_type: mime_type)
+        end
+
+        #
+        # Update image from image bytes
+        #
+        # @param [String] mime_type of image
+        # @param [Object] bytes image data
+        #
+        #
+        def update_from_bytes(bytes, mime_type: nil)
+          mime_type ||= detect_mime_from_bytes(bytes: bytes)
+          base_64_image = encode_image(mime_type: mime_type, bytes: bytes)
+          update base_64_image
+        end
+
+        #
+        # Get the mime type for the image item
+        #
+        # @return [String] mime type for image, e.g. image/png
+        #
+        def mime_type
+          state&.mime_type
+        end
+
+        #
+        # Get the bytes of the image
+        #
+        # @return [Array] Bytes that comprise the image
+        #
+        def bytes
+          state&.get_bytes
+        end
+
+        private
+
+        #
+        # Encode image information in the format required by OpenHAB
+        #
+        # @param [String] mime_type for image
+        # @param [Object] bytes image data
+        #
+        # @return [String] OpenHAB image format with image data Base64 encoded
+        #
+        def encode_image(mime_type:, bytes:)
+          "data:#{mime_type};base64,#{Base64.strict_encode64(bytes)}"
+        end
+
+        #
+        # Detect the mime type of a file
+        #
+        # @param [String] file location
+        #
+        # @return [String] mime type if it can be detected, nil otherwise
+        #
+        def detect_mime_type(file:)
+          logger.trace("Detecting mime type for file (#{file}")
+          (MimeMagic.by_path(file) || mime_from_file_contents(file: file))&.type
+        end
+
+        #
+        # Detect the mime type based on the file contents
+        #
+        # @param [String] file location
+        #
+        # @return [String] mime type if it can be detected, nil otherwise
+        #
+        def mime_from_file_contents(file:)
+          logger.trace("Detecting mime type from file (#{file} contents")
+          File.open(file) do |f|
+            MimeMagic.by_magic(f)&.type
+          end
+        end
```

Comment:
Nothing is calling / using these methods now, so they should be removed?

Reply:
removed